test/genesisProtocol.js

```diff
@@ -0,0 +1,392 @@
+import "./helpers";
+const ExecutableTest = Utils.requireContract("ExecutableTest");
+
+describe("GenesisProtocol", () => {
+  let dao, genesisProtocol, paramsHash, executableTest;
+
+  const createProposal = async function () {
+
+    const result = await genesisProtocol.propose({
+      avatar: dao.avatar.address,
+      numOfChoices: 2,
+      paramsHash: paramsHash,
+      executable: executableTest.address
+    });
+
+    assert.isOk(result);
+    assert.isOk(result.proposalId);
+
+    const stakingToken = await DAOToken.at(await genesisProtocol.stakingToken());
+
+    await stakingToken.approve(accounts[0], web3.toWei(10));
+    await stakingToken.approve(genesisProtocol.address, web3.toWei(10));
+
+    return result.proposalId;
+  };
+
+  const voteProposal = function (proposalId, how) {
+    return genesisProtocol.vote({
+      proposalId: proposalId,
+      vote: how
+    });
+  };
+
+  const stakeProposalVote = function (proposalId, how, amount) {
+    return genesisProtocol.stake({
+      proposalId: proposalId,
+      vote: how,
+      amount: web3.toWei(amount)
+    });
+  };
+
+  beforeEach(async () => {
+
+    dao = await DAO.new({
+      name: "Skynet",
+      tokenName: "Tokens of skynet",
+      tokenSymbol: "SNT",
+      schemes: [
+        { name: "GenesisProtocol" }
+      ],
+      founders: [{
+        address: accounts[0],
+        reputation: web3.toWei(1000),
+        tokens: web3.toWei(1000)
+      }
+      ]
+    });
+
+    const scheme = await dao.getSchemes("GenesisProtocol");
+
+    assert.isOk(scheme);
+    assert.equal(scheme.length, 1);
+    assert.equal(scheme[0].name, "GenesisProtocol");
+
+    genesisProtocol = await GenesisProtocol.at(scheme[0].address);
+
+    assert.isOk(genesisProtocol);
+
+    // all default parameters
+    paramsHash = (await genesisProtocol.setParams({})).result;
+
+    executableTest = await ExecutableTest.deployed();
+  });
+
+
+  it("can call getScoreThresholdParams", async () => {
+    const result = await genesisProtocol.getScoreThresholdParams({
+      avatar: dao.avatar.address,
+    });
+    assert.isOk(result);
+    assert.equal(result.thresholdConstA, 1);
+    assert.equal(result.thresholdConstB, 1);
+  });
+
+  it("can call shouldBoost", async () => {
+    const proposalId = await createProposal();
+    const result = await genesisProtocol.shouldBoost({
+      proposalId: proposalId
+    });
+    assert.equal(result, false);
+  });
+
+
+  it("can call getVoteStake", async () => {
+    const proposalId = await createProposal();
+
+    await stakeProposalVote(proposalId, 1, 10);
+
+    const result = await genesisProtocol.getVoteStake({
+      proposalId: proposalId,
+      vote: 1
+    });
+    assert.equal(web3.fromWei(result), 10);
+  });
+
+  it("can call getVotesStatus", async () => {
+    const proposalId = await createProposal();
+    const result = await genesisProtocol.getVotesStatus({
+      proposalId: proposalId
+    });
+    assert.isOk(result);
+  });
```

Comment:
What the purpose of this test ?
How about do a vote and then check the vote status ?

Reply:
The purpose is to confirm that we can call the handler method without reversions, invalid opcodes, etc.  There already exist tests such as you suggested.